src/adam_core/orbits/query/neocc.py

```diff
@@ -0,0 +1,260 @@
+
+    Parameters
+    ----------
+    object_ids : Union[List, npt.ArrayLike]
+        Object IDs / designations recognizable by NEOCC.
+    orbit_type : ["ke", "eq"]
+        Type of orbital elements to query.
+    orbit_epoch : ["middle", "present-day"]
+        Epoch of the orbital elements to query.
+
+    Returns
+    -------
+    orbits : `~adam_core.orbits.Orbits`
+        Orbits object containing the orbital elements of the specified NEOs.
+    """
+    base_url = "https://neo.ssa.esa.int/PSDB-portlet/download"
+
+    if orbit_type == "eq":
+        raise NotImplementedError("Equinoctial elements are not supported yet.")
+
+    if orbit_epoch == "middle":
+        orbit_epoch = 0
+    elif orbit_epoch == "present-day":
+        orbit_epoch = 1
+    else:
+        raise ValueError(f"Invalid orbit epoch: {orbit_epoch}")
+
+    orbits = Orbits.empty()
+
+    for object_id in object_ids:
+
+        # Clean object ID so that there are no spaces
+        object_id = object_id.replace(" ", "")
+
+        try:
+            params = {"file": f"{object_id}.{orbit_type}{orbit_epoch}"}
+
+            response = requests.get(base_url, params=params)
+            response.raise_for_status()
+
+            data = _parse_oef(response.text)
+            if orbit_type == "ke":
+
+                time_scale = data["time_system"]
+                if time_scale == "TDT":
+                    time_scale = "TT"
+                else:
+                    raise ValueError(f"Unsupported time scale: {time_scale}")
+
+                if data["header"]["refsys"] != "ECLM J2000":
+                    raise ValueError(
+                        f"Unsupported reference system: {data['header']['refsys']}"
+                    )
+
+                orbit = Orbits.from_kwargs(
+                    orbit_id=[data["object_id"]],
+                    object_id=[data["object_id"]],
+                    coordinates=KeplerianCoordinates.from_kwargs(
+                        a=[data["elements"]["a"]],
+                        e=[data["elements"]["e"]],
+                        i=[data["elements"]["i"]],
+                        raan=[data["elements"]["node"]],
+                        ap=[data["elements"]["peri"]],
+                        M=[data["elements"]["M"]],
+                        time=Timestamp.from_mjd([data["epoch"]], scale=time_scale),
+                        covariance=CoordinateCovariances.from_matrix(
+                            data["covariance"].reshape(
+                                1,
+                                6,
+                                6,
+                            )
+                        ),
+                        frame="ecliptic",
+                        origin=Origin.from_kwargs(code=["SUN"]),
+                    ).to_cartesian(),
+                )
+                orbits = qv.concatenate([orbits, orbit])
+
+        except requests.RequestException as e:
```

Comment:
I don't think we should capture the exception here. Let the caller decide what they want to do with a failed query. It will be more obvious than forgetting to handle the None and looking at the logs.